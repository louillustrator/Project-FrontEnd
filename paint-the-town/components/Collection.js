import React, { Component } from "react";
import {
  Text,
  View,
  Image,
  StyleSheet,
  ScrollView,
  ActivityIndicator,
  TouchableOpacity
} from "react-native";
import { _ } from "lodash";
import ImageFromCollection from "./ImageFromCollection";
import config from "../config";
import firebase from "firebase";
firebase.initializeApp(config);

class Collection extends Component {
  state = {
    images: {}
  };

  render() {
    const { navigate } = this.props.navigation;

    return (
      <View style={styles.view}>
        <Text style={styles.navBarText}>My Gallery</Text>
        {Object.keys(this.state.images).length === 0 ? (
          <ActivityIndicator
            size={70}
            color="#fa8231"
            animating={Object.keys(this.state.images).length === 0}
          />
        ) : (
          <ScrollView contentContainerStyle={styles.container}>
            {_.map(Object.values(this.state.images), imageLink => {
              return (
                <TouchableOpacity
                  key={imageLink._id}
                  onPress={() =>
                    navigate("ImageFromCollection", { img: imageLink })
                  }
                >
                  <Image
                    style={styles.image}
                    source={{ uri: `${imageLink.img}` }}
                  />
                </TouchableOpacity>
              );
            })}
          </ScrollView>
        )}
      </View>
    );
  }
  componentDidMount() {
    this.retrievePics().then(images => {
      this.setState({ images });
    });
  }

  retrievePics = async () => {
    //we will need to check which user is logged in for this
    const user =
      (this.props.currentUser && this.props.currentUser.username) ||
      "aDifferentTaraTest";

    let images = await firebase
      .database()
      .ref(user)
      .once("value")
      .then(function(snapshot) {
        return snapshot.val();
      });
    return images;
  };
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",

    justifyContent: "flex-start",
    flexWrap: "wrap",
    backgroundColor: "#fff"
  },
  text: {
    alignSelf: "center"
  },
  image: {
    margin: 10,

    width: 160,
    height: 160
  },

  navBarText: {
    backgroundColor: "#63cdda",
    color: "white",
    fontSize: 16,
    fontWeight: "700",
    textAlign: "center",
    paddingTop: 20,
    paddingBottom: 10
  },
  horizontal: {
    flexDirection: "row",
    justifyContent: "space-around",
    padding: 10
  }
});

export default Collection;
