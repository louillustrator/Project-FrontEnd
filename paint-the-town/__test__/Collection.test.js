import React from "react";
import { mount } from "enzyme";
import { View, TouchableHighlight } from "react-native";
import renderer from "react-test-renderer";
import Collection from "../components/Collection";

describe('Polylines', () => {
    let props;
    let state;
    beforeEach(() => {
        props = {
            navigation: {
                navigate: jest.fn()
            },
            screenProps: {
                currentUser: undefined,
                updateUser: undefined
            }
        };
        state = {
            images: null,
            loading: true
        };
    });
    it('renders without crashing', () => {
        mount(<Collection {...props} />);
    });
    it("renders correctly", () => {
        const tree = renderer.create(<Collection {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
