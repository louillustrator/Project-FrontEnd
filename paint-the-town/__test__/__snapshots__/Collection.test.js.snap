// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Polylines renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#63cdda",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "700",
        "paddingBottom": 10,
        "paddingTop": 20,
        "textAlign": "center",
      }
    }
  >
    My Gallery
  </Text>
  <ActivityIndicator
    animating={true}
    color="#fa8231"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
